packages/jellyfish-transaction-builder/__tests__/txn/txn_builder_loan_place_auction_bid.test.ts

```diff
@@ -0,0 +1,893 @@
+
+  await bob.poolpair.swap({
+    from: bobColAddr,
+    tokenFrom: 'DFI',
+    amountFrom: 600,
+    to: bobColAddr,
+    tokenTo: 'TSLA'
+  })
+  await bob.generate(1)
+  await tGroup.waitForSync()
+
+  // increase TSLA price
+  await alice.rpc.oracle.setOracleData(oracleId, timestamp, { prices: [{ tokenAmount: '15@TSLA', currency: 'USD' }] })
+  await alice.generate(1) // interest * 5 => 1000.00285385@TSLA
+
+  // check vault status before liquidated
+  const vaultBefore = await bob.container.call('getvault', [bobVaultId])
+  expect(vaultBefore.state).toStrictEqual('active')
+  expect(vaultBefore.collateralAmounts).toStrictEqual(['10000.00000000@DFI', '1.00000000@BTC'])
+  expect(vaultBefore.loanAmounts).toStrictEqual(['1000.00285385@TSLA'])
+  expect(vaultBefore.interestAmounts).toStrictEqual(['0.00285385@TSLA'])
+  expect(vaultBefore.collateralValue).toStrictEqual(20000)
+  expect(vaultBefore.loanValue).toStrictEqual(2000.0057077)
+  expect(vaultBefore.interestValue).toStrictEqual(0.0057077)
+  expect(vaultBefore.collateralRatio).toStrictEqual(1000)
+  expect(vaultBefore.informativeRatio).toStrictEqual(999.99714615)
+
+  // *6 => 1000.00342462@TSLA
+  // *7 => 1000.00399539@TSLA
+  // *8 => 1000.00456616@TSLA
+  // *9 => 1000.00513693@TSLA
+  {
+    await bob.generate(5) // *10 => 1000.0057077@TSLA
+    const vault = await bob.container.call('getvault', [bobVaultId])
+    expect(vault.collateralAmounts).toStrictEqual(['10000.00000000@DFI', '1.00000000@BTC'])
+    expect(vault.loanAmounts).toStrictEqual(['1000.00570770@TSLA'])
+    expect(vault.interestAmounts).toStrictEqual(['0.00570770@TSLA'])
+    expect(vault.collateralValue).toStrictEqual(20000)
+    expect(vault.loanValue).toStrictEqual(2000.0114154)
+    expect(vault.interestValue).toStrictEqual(0.0114154)
+    expect(vault.collateralRatio).toStrictEqual(1000)
+    expect(vault.informativeRatio).toStrictEqual(999.99429233)
+  }
+
+  const auctionsBefore = await alice.container.call('listauctions')
+  expect(auctionsBefore.length).toStrictEqual(0)
+
+  // *11 => 1000.00627847@TSLA
+  // *12 => 1000.00684924@TSLA
+  // *13 => 1000.00742001@TSLA
+  // *14 => 1000.00799078@TSLA
+  await bob.generate(6)
+  await tGroup.waitForSync()
+
+  // vault is liquidated now
+  const vaultAfter = await bob.container.call('getvault', [bobVaultId])
+  expect(vaultAfter.state).toStrictEqual('inLiquidation')
+  expect(vaultAfter.collateralAmounts).toStrictEqual(undefined)
+  expect(vaultAfter.loanAmounts).toStrictEqual(undefined)
+  expect(vaultAfter.collateralValue).toStrictEqual(undefined)
+  expect(vaultAfter.loanValue).toStrictEqual(undefined)
+  expect(vaultAfter.liquidationHeight).toStrictEqual(168)
+  expect(vaultAfter.liquidationPenalty).toStrictEqual(5)
+  expect(vaultAfter.batches).toStrictEqual([
+    { index: 0, collaterals: ['5000.00000000@DFI', '0.50000000@BTC'], loan: '500.00399539@TSLA' },
+    { index: 1, collaterals: ['5000.00000000@DFI', '0.50000000@BTC'], loan: '500.00399539@TSLA' }
```

Comment:
Can make made second batch different amount, smaller, instead of identical.

Reply:
hmm.. this part is little tricky as the collateral will be equally distributed... need huge batches only can see the diff
here:
https://github.com/DeFiCh/jellyfish/blob/44d2c03cff8621ce8b43ee9ba423f5d0a5db9e08/packages/jellyfish-transaction-builder/__tests__/txn/txn_builder_loan_place_auction_bid.test.ts#L683-L686

Reply:
btw
deposit 10k@dfi + 0.2@btc
takeloan 3k@loan
that can make the second batch smaller